tests/linearizability/watch.go

```diff
@@ -82,28 +82,37 @@ func validateWatchResponses(t *testing.T, responses [][]watchResponse, expectPro
 }
 
 func validateMemberWatchResponses(t *testing.T, responses []watchResponse, expectProgressNotify bool) {
-	var gotProgressNotify = false
-	var lastRevision int64 = 1
+	var (
+		gotProgressNotify       = false
+		lastEventRevision int64 = 1 // The event.Kv.ModRevision in the latest event.
+		lastHeadRevision  int64 = 1 // The resp.Header.Revision in last watch response.
+	)
 	for _, resp := range responses {
-		if resp.Header.Revision < lastRevision {
-			t.Errorf("Revision should never decrease")
+		if resp.Header.Revision < lastEventRevision || resp.Header.Revision < lastHeadRevision {
+			t.Errorf("Revision should never decrease[1], lastEventRevision: %d, lastHeadRevision: %d, resp.Header.Revision: %d",
+				lastEventRevision, lastHeadRevision, resp.Header.Revision)
 		}
-		if resp.IsProgressNotify() && resp.Header.Revision == lastRevision {
+
+		if resp.IsProgressNotify() && resp.Header.Revision == lastHeadRevision {
 			gotProgressNotify = true
 		}
-		if resp.Header.Revision == lastRevision && len(resp.Events) != 0 {
+		if resp.Header.Revision == lastHeadRevision && len(resp.Events) != 0 {
 			t.Errorf("Got two non-empty responses about same revision")
 		}
+
 		for _, event := range resp.Events {
-			if event.Kv.ModRevision != lastRevision+1 {
-				t.Errorf("Expect revision to grow by 1, last: %d, mod: %d", lastRevision, event.Kv.ModRevision)
+			if event.Kv.ModRevision != lastEventRevision+1 {
+				t.Errorf("Expect revision to grow by 1, last: %d, mod: %d", lastEventRevision, event.Kv.ModRevision)
 			}
-			lastRevision = event.Kv.ModRevision
+			lastEventRevision = event.Kv.ModRevision
 		}
-		if resp.Header.Revision != lastRevision {
-			t.Errorf("Expect response revision equal last event mod revision")
+
+		if resp.Header.Revision < lastEventRevision || resp.Header.Revision < lastHeadRevision {
```

Comment:
'resp.Header.Revision < lastHeadRevision' | could this be successful in line 91 and fail here ?

Reply:
NO. I am aware of that it, since it isn't a big deal, so I did not remove the duplicated if condition. 

But updated anyway & thx